uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js

```diff
@@ -293,6 +290,33 @@ define([
     }
 
     /**
+     * Get status of the document based on the attachments count
+     * @return {string|int}
+     */
+    function getDocumentStatus () {
+      return vm.isRole('admin')
+        ? (vm.files.length || vm.uploader.queue.length ? vm.getStatusIdByName('approved') : vm.getStatusIdByName('awaiting upload'))
+        : (vm.files.length || vm.uploader.queue.length ? vm.getStatusIdByName('awaiting approval') : vm.getStatusIdByName('awaiting upload'));
+    }
+
+    /**
+     * Gets the status id of a documents status based on it name
+     * @param  {string} statusName
+     * @return {string|int}
+     */
+    function getStatusIdByName (statusName) {
+      var statusId = '';
+
+      _.find($rootScope.cache.documentStatus.obj, function (status, id) {
```

Comment:
This is not how `find` is supposed to work. It should be something like this
```
var status = _.find($rootScope.cache.documentStatus.obj, function (status) {
	return status.name === statusName;
});

return status ? status.id : false;
```

Reply:
The following works:
```js
function getStatusIdByName (statusName) {
      return _.findKey($rootScope.cache.documentStatus.obj, function(status) {
        return status === statusName;
      });
    }
```